blockly/src/antlr/BlocklyConditionVisitor.java

```diff
@@ -203,7 +209,109 @@ public INode visitFunc_call(blocklyParser.Func_callContext ctx) {
+              yield false;
+            }
+
+            // Get the first argument and visit it
+            INode argNode = visit(ctx.args.expr(0));
+            String direction;
+
+            if (argNode instanceof BaseNode && ((BaseNode) argNode).baseType == Types.STRING) {
+              direction = ((BaseNode) argNode).strVal;
+            } else {
+              LOGGER.warning("Expected string argument for nahePit");
+              yield false;
+            }
+
+            yield httpServer.isNearTile(PitTile.class, Direction.fromString(direction));
+          }
+          case "naheMonster" -> {
+            // Check if arguments exist
+            if (ctx.args == null) {
+              LOGGER.warning("naheMonster function: Expected 1 argument, got 0");
+              yield false;
+            }
+
+            // Get the first argument and visit it
+            INode argNode = visit(ctx.args.expr(0));
+            String direction;
+
+            if (argNode instanceof BaseNode && ((BaseNode) argNode).baseType == Types.STRING) {
+              direction = ((BaseNode) argNode).strVal;
+            } else {
+              LOGGER.warning("Expected string argument for naheMonster");
+              yield false;
+            }
+
+            yield httpServer.isNearComponent(
+                BlocklyMonsterComponent.class, Direction.fromString(direction));
+          }
+          case "naheSchalter" -> {
+            // Check if arguments exist
+            if (ctx.args == null) {
+              LOGGER.warning("naheSchalter function: Expected 1 argument, got 0");
+              yield false;
+            }
+
+            // Get the first argument and visit it
+            INode argNode = visit(ctx.args.expr(0));
+            String direction;
+
+            if (argNode instanceof BaseNode && ((BaseNode) argNode).baseType == Types.STRING) {
+              direction = ((BaseNode) argNode).strVal;
+            } else {
+              LOGGER.warning("Expected string argument for naheSchalter");
+              yield false;
+            }
+
+            yield httpServer.isNearComponent(LeverComponent.class, Direction.fromString(direction));
+          }
+          case "naheBrotkrume" -> {
+            // Check if arguments exist
+            if (ctx.args == null) {
+              LOGGER.warning("naheBrotkrume function: Expected 1 argument, got 0");
+              yield false;
+            }
+
+            // Get the first argument and visit it
+            INode argNode = visit(ctx.args.expr(0));
+            String direction;
+
+            if (argNode instanceof BaseNode && ((BaseNode) argNode).baseType == Types.STRING) {
+              direction = ((BaseNode) argNode).strVal;
+            } else {
+              LOGGER.warning("Expected string argument for naheBrotkrume");
+              yield false;
+            }
+
+            yield httpServer.isNearComponent(
+                BreadcrumbComponent.class, Direction.fromString(direction));
```

Comment:
Jetzt haben wir 5x die gleiche Logik nur mit unterschiedlichen Parametern. Bitte eine kleine einfache `private` Methode erstellen.

Reply:
You are right, im a copy pasta type of guy